ygen/codegen.go

```diff
@@ -712,26 +711,31 @@ func mappedDefinitions(yangFiles, includePaths []string, cfg *GeneratorConfig) (
 	// them from the modules that are provided as an argument.
 	dirs := make(map[string]*yang.Entry)
 	enums := make(map[string]*yang.Entry)
-	var rootElems []*yang.Entry
+	var rootElems, treeElems []*yang.Entry
 	for _, module := range modules {
-		findMappableEntities(module, dirs, enums, cfg.ExcludeModules, cfg.CompressOCPaths)
+		errs = append(errs, findMappableEntities(module, dirs, enums, cfg.ExcludeModules, cfg.CompressOCPaths, modules)...)
 		if module == nil {
 			errs = append(errs, errors.New("found a nil module in the returned module set"))
 			continue
 		}
 		// Ensure that we do not try and traverse an empty module.
-		if module.Dir != nil && !excluded[module.Name] {
+		if module.Dir != nil {
```

Comment:
optional: len(module.Dir) would be more robust if the semantics of that field are not explicitly defined.

Reply:
I think this check can actually be entirely removed, since we'd expect that the map in `Dir` is nil and range will still not panic with NPE here. Removing.